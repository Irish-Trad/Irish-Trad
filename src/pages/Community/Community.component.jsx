import React from 'react'

import { MainContainer } from './Community.styles'

const Community = () => <MainContainer><h1>Community</h1></MainContainer>

export default Community