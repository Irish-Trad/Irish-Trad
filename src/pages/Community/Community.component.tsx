import React from 'react'
import Metadata from 'common/Metadata'

import {
  MainContainer,
  H1,
  Text,
  ExternalLink,
  Figure,
  Figcaption
} from 'common/StyledComponents/StyledComponents.js'

import {
  StyledLink,
  StyledIcon,
  ButtonContainer,
  Button,
  InternalLink,
  Image
} from './Community.styles'

const Community: React.FunctionComponent = () => (
  <MainContainer>
    <Metadata
      title="Irish-Trad.net - Community"
      metaTitle="Irish-Trad.net - Community"
      metaKeywords="Irish Trad, Irish Traditional, Irish Tenor Banjo, Irish Bouzouki, Bouzouki, Mandolin, Octave Mandolin, Mandola, Octave Mandola, Chords, TAB, Tablature"
      metaDescription="Community for Irish-Trad.net"
    />
    <H1>Community</H1>
    <Text>
      I have created a Discord server to allow users of Irish-Trad.net to
      interact with each other and discuss various topics regarding Irish Trad
      music.
    </Text>
    <Text>
      To join the Irish-Trad.net Discord Server simply click the following Icon
      or Button:
    </Text>
    <ButtonContainer>
      <StyledLink href="https://discord.gg/Mq9jeA8" target="_blank">
        <StyledIcon icon={['fab', 'discord']} />
      </StyledLink>
      <ExternalLink href="https://discord.gg/Mq9jeA8" target="_blank">
        <Button>Click Here To Join Discord</Button>
      </ExternalLink>
    </ButtonContainer>
    <Text>
      Discord is a free platform that allows for live voice and text
      communication similar to Skype, Messenger and IRC channels of old.
    </Text>
    <Text>
      To get an idea of what a Discord server looks like you can click the
      following Image:
    </Text>
    <Figure>
      <InternalLink to="../assets/discord.jpg" target="_blank">
        <Image
          src="../assets/discord.jpg"
          alt="Screenshot of Discord"
          title="Screenshot of Discord"
        />
      </InternalLink>
      <Figcaption>Pictured: Screenshot of Irish-Trad Discord</Figcaption>
    </Figure>
    <Text>
      Clicking on one of the specific channels found on the left brings you into
      that chatroom were you can discuss that particular topic.
    </Text>
    <Text>
      For more information about Discord you can visit the official Discord
      webpage here:{' '}
      <ExternalLink
        href="https://discordapp.com/"
        target="_blank"
        rel="noopener noreferrer"
      >
        Discord Website
      </ExternalLink>
    </Text>
    <Text>
      Discord can be accessed through any web browser and downloaded as an
      application on desktop and mobile.
    </Text>
    <Text>
      The Discord icon can also be found in the footer of every page to make
      future access easier.
    </Text>
  </MainContainer>
)

export default Community
