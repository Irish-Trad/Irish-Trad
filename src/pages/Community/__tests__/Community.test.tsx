import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Community from '../Community.component'

const renderComponent = () => render(<Community />)

describe('Community', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snapshot', () => {
    const snapshot = renderComponent()

    expect(snapshot).toMatchSnapshot()
  })

  it('Title of the page is Community', () => {
    const { getByText } = renderComponent()

    expect(getByText('Community')).toBeInTheDocument()
  })
})
