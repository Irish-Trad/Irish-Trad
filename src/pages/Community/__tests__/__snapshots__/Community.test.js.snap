// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-bdVaJa aBmTA"
      >
        <h1
          class="sc-bwzfXH kWcKFk"
        >
          Community
        </h1>
        <p
          class="sc-bxivhb faZbih"
        >
          I have created a Discord server to allow users of Irish-Trad.net to interact with each other and discuss various topics regarding Irish Trad music.
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          To join the Irish-Trad.net Discord Server simply click the following Icon or Button:
        </p>
        <div
          class="sc-gZMcBi hLSPcS"
        >
          <a
            class="sc-dnqmqq bDsesi"
            href="https://discord.gg/Mq9jeA8"
            target="_blank"
          >
            <i
              class="fa"
            />
          </a>
          <a
            class="sc-htoDjs gjSkhw"
            href="https://discord.gg/Mq9jeA8"
            target="_blank"
          >
            <button
              class="sc-gqjmRU isyvva"
            >
              Click Here To Join Discord
            </button>
          </a>
        </div>
        <p
          class="sc-bxivhb faZbih"
        >
          Discord is a free platform that allows for live voice and text communication similar to Skype, Messenger and IRC channels of old.
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          To get an idea of what a Discord server looks like you can click the following Image:
        </p>
        <figure
          class="sc-ifAKCX hVeHwL"
        >
          <a
            class="sc-dnqmqq bDsesi"
            href="assets/discord.jpg"
            target="_blank"
          >
            <img
              alt="Screenshot of Discord"
              class="sc-VigVT kxLHnZ"
              src="assets/discord.jpg"
              title="Screenshot of Discord"
            />
          </a>
          <figcaption
            class="sc-bZQynM hJohEA"
          >
            Pictured: Screenshot of Irish-Trad Discord
          </figcaption>
        </figure>
        <p
          class="sc-bxivhb faZbih"
        >
          Clicking on one of the specific channels found on the left brings you into that chatroom were you can discuss that particular topic.
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          For more information about Discord you can visit the official Discord webpage here:
           
          <a
            class="sc-htoDjs gjSkhw"
            href="https://discordapp.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Discord Website
          </a>
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          Discord can be accessed through any web browser and downloaded as an application on desktop and mobile.
        </p>
        <p
          class="sc-bxivhb faZbih"
        >
          The Discord icon can also be found in the footer of every page to make future access easier.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-bdVaJa aBmTA"
    >
      <h1
        class="sc-bwzfXH kWcKFk"
      >
        Community
      </h1>
      <p
        class="sc-bxivhb faZbih"
      >
        I have created a Discord server to allow users of Irish-Trad.net to interact with each other and discuss various topics regarding Irish Trad music.
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        To join the Irish-Trad.net Discord Server simply click the following Icon or Button:
      </p>
      <div
        class="sc-gZMcBi hLSPcS"
      >
        <a
          class="sc-dnqmqq bDsesi"
          href="https://discord.gg/Mq9jeA8"
          target="_blank"
        >
          <i
            class="fa"
          />
        </a>
        <a
          class="sc-htoDjs gjSkhw"
          href="https://discord.gg/Mq9jeA8"
          target="_blank"
        >
          <button
            class="sc-gqjmRU isyvva"
          >
            Click Here To Join Discord
          </button>
        </a>
      </div>
      <p
        class="sc-bxivhb faZbih"
      >
        Discord is a free platform that allows for live voice and text communication similar to Skype, Messenger and IRC channels of old.
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        To get an idea of what a Discord server looks like you can click the following Image:
      </p>
      <figure
        class="sc-ifAKCX hVeHwL"
      >
        <a
          class="sc-dnqmqq bDsesi"
          href="assets/discord.jpg"
          target="_blank"
        >
          <img
            alt="Screenshot of Discord"
            class="sc-VigVT kxLHnZ"
            src="assets/discord.jpg"
            title="Screenshot of Discord"
          />
        </a>
        <figcaption
          class="sc-bZQynM hJohEA"
        >
          Pictured: Screenshot of Irish-Trad Discord
        </figcaption>
      </figure>
      <p
        class="sc-bxivhb faZbih"
      >
        Clicking on one of the specific channels found on the left brings you into that chatroom were you can discuss that particular topic.
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        For more information about Discord you can visit the official Discord webpage here:
         
        <a
          class="sc-htoDjs gjSkhw"
          href="https://discordapp.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Discord Website
        </a>
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        Discord can be accessed through any web browser and downloaded as an application on desktop and mobile.
      </p>
      <p
        class="sc-bxivhb faZbih"
      >
        The Discord icon can also be found in the footer of every page to make future access easier.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
